sc2/unit.py

```diff
@@ -1177,6 +1185,10 @@ def weapon_cooldown(self) -> float:
             return self._proto.weapon_cooldown
         return -1
 
+    @property
+    def weapon_ready(self) -> bool:
+        return self.weapon_cooldown < self._bot_object.client.game_step
```

Comment:
Maybe others could join the conversation, why would this be more ideal than 
`return self.weapon_cooldown <= self._bot_object.client.game_step` 
or
`return self.weapon_cooldown < self._bot_object.client.game_step / 2`.
Perhaps this only works up to a certain `client.game_step` value. Obviously it would be impossible to micro units with `client.game_step = 1000`.

Reply:
I have changed this to `self.weapon_cooldown ==0`.